compiler/rustc_codegen_ssa/src/mir/constant.rs

```diff
@@ -60,36 +61,48 @@ impl<'a, 'tcx, Bx: BuilderMethods<'a, 'tcx>> FunctionCx<'a, 'tcx, Bx> {
         self.cx.tcx().const_eval_resolve_for_typeck(ty::ParamEnv::reveal_all(), uv, constant.span)
     }
 
-    /// process constant containing SIMD shuffle indices
-    pub fn simd_shuffle_indices(
+    /// process constant SIMD vector or constant containing SIMD shuffle indices
+    pub fn early_evaluate_const_vector(
```

Comment:
I find "early" here quite confusing, I don't think we otherwise use this adjective to refer to compile-time evaluation. Why not just `evaluate_const_vector`?